packages/@sanity/cli/src/actions/init-project/initProject.ts

```diff
@@ -202,9 +226,205 @@ export default async function initSanity(
   // Ensure we are using the output path provided by user
   outputPath = answers.outputPath
 
+  if (initNext) {
+    const useTypeScript = await prompt.single<boolean>({
+      type: 'confirm',
+      message: `Would you like to use TypeScript?`,
+      default: true,
+    })
```

Comment:
If I'm not mistaken there's already a prompt that you can use `promptForTypeScript` over on `prompts.js`

I'm unsure of the flow of the CLI (so it might be in a different path in which case please ignore this last bit) but you might also be able to use `useTypeScript` in line 225?